staging/src/k8s.io/dynamic-resource-allocation/cel/compile.go

```diff
@@ -59,6 +68,8 @@ func newCompiler() *compiler {
 }
 
 // CompileCELExpression returns a compiled CEL expression. It evaluates to bool.
+//
+// TODO: validate AST to detect invalid attribute names.
```

Comment:
For example, `device.attributes["foo_bar"]` is valid CEL but cannot be right because `foo_bar` is not a DNS sub-domain and thus can never match a device attribute.

IMHO not needed in 1.31. We may also want to add new status fields to report problems with CEL evaluation, which is definitely a future enhancement.

Reply:
might be worth updating the comment to include a link to the issue for more details

make sure whatever increased validation is done will do something sensible if already-persisted expressions that would not pass the validation is encountered

Reply:
> might be worth updating the comment to include a link to the issue for more details

Added. Will be in the next PR update.